packages/expo-analytics-amplitude/src/Amplitude.ts

```diff
@@ -0,0 +1,51 @@
+import { UnavailabilityError } from 'expo-errors';
+import ExpoAmplitude from './ExpoAmplitude';
+
+export function initialize(apiKey: string): Promise<void> {
```

Comment:
it seems to be bad that a function returning a promise can throw exceptions out of the promise scope 🤔 I mean the unavailability errors won't be passed to promise.catch, but will be thrown in the synchronous way. Maybe we should make all these functions `async`?

Reply:
As per the discussion, let's leave that this way 🙂